### PR TITLE
ReadableStreamDefaultController.close() - invalid exception reasons

### LINKED ISSUE #34597: Error reasons not documented
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/API/ReadableStreamDefaultController/close

### What specific section or headline is this issue about?

Exceptions

### What information was incorrect, unhelpful, or incomplete?

The "Exceptions" section says:

> [TypeError](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypeError)
>   Thrown if the source object is not a ReadableStreamDefaultController.

This suggests that errors are not thrown in other cases.

### What did you expect to see?

I expect to see that a `TypeError` may be thrown by this method if the stream is already in the process of closing or closed.

### Do you have any supporting links, references, or citations?

https://streams.spec.whatwg.org/#rs-default-controller-close

### Do you have anything more you want to share?

Similar issues exist for `ReadableStreamDefaultController.enqueue` and probably methods on `WritableStreamDefaultController` as well.

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/api/readablestreamdefaultcontroller/close`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/ReadableStreamDefaultController/close
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/api/readablestreamdefaultcontroller/close/index.md
* Last commit: https://github.com/mdn/content/commit/d8b4431bfde42f1bc195239ea1f378d763f8163e
* Document last modified: 2024-04-22T06:16:40.000Z

</details>

### PR DESCRIPTION
**Pull request by hamishwillee**
Partial fix for  #34597

Also some layout only fixes.

### CODE REVIEW

File: files/en-us/web/api/readablestreamdefaultcontroller/close/index.md [hunk 1]
```diff
@@ -34,19 +29,15 @@ None ({{jsxref("undefined")}}).
 ### Exceptions
 
 - {{jsxref("TypeError")}}
-  - : Thrown if the source object is not a `ReadableStreamDefaultController`.
+
+  - : Thrown if `close()` is called when the stream is not readable — because it is already closed, cancelled, or errored — or because it has been requested to close by the underlying source but it has not yet done so because there are still enqueued chunks to read.
```
Comment (hamishwillee) [inline files/en-us/web/api/readablestreamdefaultcontroller/close/index.md:33]: This is the only content change.

**Comment by Josh-Cena** [comment]
Is this intended to close #34597?

**Comment by hamishwillee** [comment]
@Josh-Cena No, it only partially fixes it. The original comment states that there the same problem is found in some nearby issues. I didn't look at those, though I will later if no one else does.

**Comment by Josh-Cena** [comment]
Ok; because the PR description containing the substring "fixes #xxx", even if prefixed by "partially", would link this PR.

**Comment by hamishwillee** [comment]
Thanks @Josh-Cena - I didn't realize it was that sensitive. Have modified to something that hopefully wouldn't trigger a close.

**Comment by hamishwillee** [comment]
@wbamberg When you have a moment, this one should be an easy review. The only actual content change is highlighted above https://github.com/mdn/content/pull/34690/#pullrequestreview-2161961615

**Review by wbamberg — APPROVED** [review]
👍

**Pull request merged**